src/app/plan/components/Wizard/VolumesForm.tsx

```diff
@@ -34,30 +52,119 @@ const VolumesForm = props => {
     }
   }, []);
 
+  const discoveredPersistentVolumes = (currentPlan && currentPlan.spec.persistentVolumes) || [];
+
+  useEffect(() => {
+    if (discoveredPersistentVolumes.length > 0) {
+      getPVResourcesRequest(discoveredPersistentVolumes, values.sourceCluster || '');
+      let mappedPVs;
+      if (values.persistentVolumes) {
+        mappedPVs = discoveredPersistentVolumes.map(planVolume => {
+          let pvAction = 'copy'; // Default to copy
+          if (values.persistentVolumes.length !== 0) {
+            const rowVal = values.persistentVolumes.find(v => v.name === planVolume.name);
+            if (rowVal && rowVal.selection) {
+              pvAction = rowVal.selection.action;
+            } else {
+              if (rowVal && rowVal.type) {
+                pvAction = rowVal.type;
+              }
+            }
+          }
+          // TODO do we really need to generate these meta-objects and keep them in formik?
+          // currentPlan.spec.persistentVolumes is in global state, so we could just
+          // infer from that everywhere and keep only a mapping of pv id => type selections in formik
+          return {
+            name: planVolume.name,
+            project: planVolume.pvc.namespace,
+            storageClass: planVolume.storageClass || 'None',
+            size: planVolume.capacity,
+            claim: planVolume.pvc.name,
+            type: pvAction,
+            details: '',
+            supportedActions: planVolume.supported.actions,
+          };
+        });
+      } else {
+        mappedPVs = discoveredPersistentVolumes.map(planVolume => {
+          const pvAction = 'copy'; // Default to copy
+          return {
+            name: planVolume.name,
+            project: planVolume.pvc.namespace,
+            storageClass: planVolume.storageClass || '',
+            size: planVolume.capacity,
+            claim: planVolume.pvc.name,
+            type: pvAction,
+            details: '',
+            supportedActions: planVolume.supported.actions,
+          };
+        });
+      }
+      setFieldValue('persistentVolumes', mappedPVs);
+    }
+  }, [discoveredPersistentVolumes.length]); // Only re-run the effect if fetching value changes
```

Comment:
+1 Nice job with the useEffect enhancements. Originally we were using useEffect to filter out all of the polling updates in the background causing rerenders. Glad this is getting some attention! Looks much better.

Reply:
Ah that makes sense. Unnecessary re-renders are still a thing I never know how to handle correctly.